sdks/router-sdk/src/utils/encodeMixedRouteToPath.ts

```diff
@@ -29,9 +29,30 @@ export function encodeMixedRouteToPath(route: MixedRouteSDK<Currency, Currency>)
     path = [route.pathInput.isNative ? ADDRESS_ZERO : route.pathInput.address]
     types = ['address']
     let currencyIn = route.pathInput
+    let lastCurrencyOut = undefined
 
-    for (const pool of route.pools) {
-      const currencyOut = currencyIn.equals(pool.token0) ? pool.token1 : pool.token0
+    for (let i = 0; i < route.pools.length; i++) {
+      const pool = route.pools[i]
+      // it's possible lastCurrencyOut is native and currencyIn is wrapped native, or vice versa
+      // in this case we need to wrap both and check they are equal
+      const currencyOut = currencyIn.wrapped.equals(pool.token0.wrapped) ? pool.token1 : pool.token0
+
+      if (lastCurrencyOut) {
+        const lastCurrencyOutNativeCurrencyInWrapped = lastCurrencyOut.isNative && currencyIn.isToken && lastCurrencyOut.wrapped.equals(currencyIn)
```

Comment:
aren't lastCurrencyOut and currencyIn equal to each other? 
should it be lastCurrencyOut and currencyOut?